aws/src/main/java/org/apache/iceberg/aws/s3/AnalyticsAcceleratorUtil.java

```diff
@@ -27,71 +27,59 @@
 import org.apache.iceberg.util.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import software.amazon.awssdk.services.s3.S3AsyncClient;
-import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.S3Client;
 import software.amazon.s3.analyticsaccelerator.ObjectClientConfiguration;
-import software.amazon.s3.analyticsaccelerator.S3SdkObjectClient;
 import software.amazon.s3.analyticsaccelerator.S3SeekableInputStream;
 import software.amazon.s3.analyticsaccelerator.S3SeekableInputStreamConfiguration;
 import software.amazon.s3.analyticsaccelerator.S3SeekableInputStreamFactory;
+import software.amazon.s3.analyticsaccelerator.S3SyncSdkObjectClient;
 import software.amazon.s3.analyticsaccelerator.common.ConnectorConfiguration;
 import software.amazon.s3.analyticsaccelerator.request.ObjectClient;
-import software.amazon.s3.analyticsaccelerator.request.ObjectMetadata;
-import software.amazon.s3.analyticsaccelerator.util.OpenStreamInformation;
 import software.amazon.s3.analyticsaccelerator.util.S3URI;
 
 class AnalyticsAcceleratorUtil {
 
   private static final Logger LOG = LoggerFactory.getLogger(AnalyticsAcceleratorUtil.class);
 
-  private static final Cache<Pair<S3AsyncClient, S3FileIOProperties>, S3SeekableInputStreamFactory>
+  private static final Cache<Pair<S3Client, S3FileIOProperties>, S3SeekableInputStreamFactory>
       STREAM_FACTORY_CACHE =
           Caffeine.newBuilder()
               .maximumSize(100)
               .removalListener(
                   (RemovalListener<
-                          Pair<S3AsyncClient, S3FileIOProperties>, S3SeekableInputStreamFactory>)
+                          Pair<S3Client, S3FileIOProperties>, S3SeekableInputStreamFactory>)
                       (key, factory, cause) -> close(factory))
               .build();
 
   private AnalyticsAcceleratorUtil() {}
 
   public static SeekableInputStream newStream(S3InputFile inputFile) {
     S3URI uri = S3URI.of(inputFile.uri().bucket(), inputFile.uri().key());
-    HeadObjectResponse metadata = inputFile.getObjectMetadata();
-    OpenStreamInformation openStreamInfo =
-        OpenStreamInformation.builder()
-            .objectMetadata(
-                ObjectMetadata.builder()
-                    .contentLength(metadata.contentLength())
-                    .etag(metadata.eTag())
-                    .build())
-            .build();
 
     S3SeekableInputStreamFactory factory =
         STREAM_FACTORY_CACHE.get(
-            Pair.of(inputFile.asyncClient(), inputFile.s3FileIOProperties()),
+            Pair.of(inputFile.client(), inputFile.s3FileIOProperties()),
```

Comment:
dont we need to make a decision here to get sync or async client based on S3FileIOProperties? Or, will InputFile return the right one?